### DISCUSSION TITLE
classes/class_audiostreamplaylist

### DISCUSSION THREAD

**Discussion by giscus** [godotengine/godot-docs-user-notes#281] [User-contributed notes]
# classes/class_audiostreamplaylist

Inherits: AudioStream< Resource< RefCounted< Object AudioStream that includes sub-streams and plays them back like a playlist. Properties:,,, float, fade_time, 0.3,, bool, loop, true,, bool, shuffl...

https://docs.godotengine.org/en/stable/classes/class_audiostreamplaylist.html

**Comment by Kartopod** [comment]
How do you get the index of the stream that is currently playing?

**Reply by HephepTeam** [reply]
How do you get the list of streams :(

**Reply by malexit240** [reply]
Try to use cycle. There is property stream_count (list length) and method get_list_stream (one by index)

**Reply by kiebor81** [reply]
Just attach this to the audio stream player

```gdscript
func get_current_stream_name():
	var playlist = stream as AudioStreamPlaylist
	if playlist:
		var position = get_playback_position()
		var cumulative_time = 0.0
		
		for i in range(playlist.get_stream_count()):
			var _stream = playlist.get_list_stream(i)
			
			if _stream:
				cumulative_time += _stream.get_length()
				if position < cumulative_time:
					return _stream.resource_path

	return "No stream playing"
```

**Comment by udeveloping** [comment]
How do i use : void set_list_stream(stream_index: int, audio_stream: AudioStream) ????

I wanna manage all my game music from a AudioStreamPlayer with a Playlist resource putting it as a Singletone node, but i can't figure out how to use the set_list_stream to change the song into my Playlist resource. Regards
